src/com/yashu/interview/map/linkedhashset/HashSet.java

```diff
@@ -0,0 +1,218 @@
+package HashSet;
+import java.util.InputMismatchException;
+public class HashSet<T> implements OrderedSet<T>, Container<T>{
+
+	private	Node<T>[] array;
+	private ArrayIndexCalculator indexCalculator;
+	private LinkedList<T> linkedList;
+	private int count = 0;
+
+
+	/**
+	 * Parameterized constructor
+	 * @param initialSize - Initial size of the set
+	 */
+	@SuppressWarnings("unchecked")
+	public HashSet(int initialSize){
+		array = new Node[initialSize];
+		indexCalculator = new ArrayIndexCalculator();
+		linkedList = new LinkedList<T>();
+	}
+	 
+	// Package visible only - for testing purposes
+	@SuppressWarnings("unchecked")
+	HashSet(int initialSize, ArrayIndexCalculator indexCalculator) {
+		array = new Node[initialSize];
+		this.indexCalculator = indexCalculator;
+		linkedList = new LinkedList<T>();
+	}
+	// Returns the size of the Array.
+	public int getLengthOfArray() {
+		return array.length;
+	}
+
+	@SuppressWarnings("unchecked")
+	private void createNewArray(int length) {
+		array = new Node[length];
+	}
+	
+	//Iterator
+	private class LinkedHashSetIterator implements MyIterator<T>{
+		private Node<T> temp ;
+		
+		public LinkedHashSetIterator(){
+		temp = null;
+		}
+		@Override
+		public boolean hasNext() {
+			return temp!=linkedList.getTail();
+		}
+
+		@Override
+		public Object next() {
+			if(temp == null){
+				temp = linkedList.getHead() ;
+				return temp.getData();
+			}
+			temp = temp.getNext();
+			if(this.hasNext()){
+				return temp.getData();
+			}
+			return temp.getData();
+		}
+		
+	}
+	
+	@Override
+	public MyIterator<T> iterator() {
+		return new LinkedHashSetIterator();
+	}
+	
+	public void insert(T value) {
+		if(value == null){
+			throw new IllegalArgumentException();
+		}
+		//Checks uniqueness of the set.
+		if(exists(value)){return;}
+		//Doubles the size of Array if it is full.
+		if(count == array.length){
+			array = doubleArray(array.length);
+		}
+		count++;
+		Node<T> newNode = new Node<T>(value);
+		//Using the Hash Function the index of the node is determined.
+		int indexOfString = hashFunction(value);
+		//If the index is not null, collision is avoided using LinkedList. 
+		if(collision(value)){
+			linkedList.setTail(avoidCollision(value,newNode,linkedList.getTail(),indexOfString));
+			count--;
```

Comment:
If you are only tracking the count of elements inserted in the array, excluding the collided entries, and double array only when it gets full, HashSet performance will be inefficient.  It is very likely that, it will never get full but, lots entries keep getting added to the collision linked list. 

To double the array take into account, total number of elements inserted.